pkg/detectors/detectors.go

```diff
@@ -59,7 +61,42 @@ type Result struct {
 
 	// This field should only be populated if the verification process itself failed in a way that provides no
 	// information about the verification status of the candidate secret, such as if the verification request timed out.
-	VerificationError error
+	verificationError error
+}
+
+// SetVerificationError is the only way to set a verification error. Any sensetive values should be passed-in as secrets to be redacted.
+func (r *Result) SetVerificationError(err error, secrets ...string) {
+	if err != nil {
+		r.verificationError = redactSecrets(err, secrets...)
+	}
+}
+
+// Public accessors for the fields could also be provided if needed.
+func (r *Result) VerificationError() error {
+	return r.verificationError
+}
+
+// redactSecrets replaces all instances of the given secrets with [REDACTED] in the error message.
+func redactSecrets(err error, secrets ...string) error {
+	lastErr := unwrapToLast(err)
+	errStr := lastErr.Error()
```

Comment:
Since you're throwing away all the intermediate errors eventually anyway, could you just stringify `err` without worrying about unwrapping it yourself, and then do the string replacement? I'm sure what you've got works, but verifying that requires knowing the details of how error unwrapping works, whereas if you just stringify from the outset you don't need to think about any of that.

Reply:
that's what i did first. but in testing this with @trufflesteeeve we realized that sometimes the errors are super long and contain information that isn't pertinent - URLs and such, for example. And since it's generally the last error that has the most value, this ensures the logs output is compact, reduces chances of errors containing secrets which helps with performance somewhat in redaction.

Reply:
Ah, ok, that makes sense. Would you mind putting it in a code comment?